### PR TITLE
Icons page redesign

### LINKED ISSUE #1687: Update Icons page on website with updated icons and recommentations
Update the Icons page to reflect the PatternFly icon and usage recommendations found in [this  spreadsheet](https://docs.google.com/spreadsheets/d/1FwcP3qQm6zseNT-MKM2BwEmlu1La6KSmWe7i-9JzNFw/edit?usp=sharing): 

- **icons in yellow (12)** have open questions. These icons have notes, some of which instruct to keep the usage text blank or leave the icon off the webpage (until we get more info from the team)
- **icons in pink (9)** these will be added to a migration table. These icons have notes, some of which instruct to leave the icon off of the webpage (until we create the icon migration table), but the new icon recommendations will remain on the webpage and are marked as such.
- **icons in blue (4)** need to be added to the PF font. There is a separate story for adding them here: https://github.com/patternfly/patternfly-next/issues/2707
- any new usage text is prepended with "Suggestion:" Any usage text with a strikethrough should be left off of the webpage for now (until we get more info from the team)

Columns to display: Icon Usage, Icon, Icon Name, Icon Type, Font Set

The original icon story is here: [https://github.com/patternfly/patternfly-design/issues/787](https://github.com/patternfly/patternfly-design/issues/787)

Work with @bdellasc throughout the process

cc @rachael-phillips @LHinson

### PR DESCRIPTION
**Pull request by evwilkin**
Closes #1687 

This PR rebuilds the icons page based on redesign & updated icon recommendations.

### CODE REVIEW

**Comment by redallen** [comment]
Rebased on `master` since `v4` is about to be deleted.

**Comment by mcoker** [comment]
Under "small icons", the dropdown toggle example - is the down arrow supposed to be small? If so, it's 16x16.

The App launcher menu beside ^ dropdown is a `<nav>` element (when it isn't a `<nav>` for the page) and the spacing is wrong because there are 2 `.pf-c-app-launcher__group` elements in the menu, and the first one is hidden via `.ws-icon-size-examples .ws-icons-appLauncher .pf-c-app-launcher__group:first-child, .ws-icon-size-examples .ws-icons-appLauncher .pf-c-app-launcher__toggle { display: none; }`. We rely on the `:first-child` section to have custom spacing for the menu to display properly.

Under "medium icons" the alert has an `<h4>` title that isn't a section title for the page.

Under "x-large icons" the empty state has an `<h5>` with the same issue.

Also, the presentation of the small/medium/large/x-large icons box where we include these examples seems a little odd. One of the examples is a screenshot. IMO, they should all be screenshots. That would give you more control over their presentation and you won't end up with the problem of those examples having invalid markup for their role on the page. Here's what it looks like when the nav is hidden, since they're regular html/css and are responsive. It also shows how we need some space between the alignment screenshot and this example box.

Screen Shot 2020-05-20 at 10 14 14 AM

And here is the screen right before the nav is hidden, showing the small/medium/large/x-large example box overlapping the alignment image in the left column.

Screen Shot 2020-05-20 at 10 15 05 AM


> Use this for 'fa' regular icons: <i className="far [insert-icon-name]"></i>

A user would have to bundle their own font-awesome regular package for this to work since we only ship the font-awesome solid webfont. Is that noted anywhere? 

In the "Updated icon recommendations" table, the way the names wrap is kinda hard to read. It also doesn't present the same as the "all icons" table above it. If we kept the icon/name cells from wrapping and made it a compact table, I think it would work a lot better.

#### Old:

Screen Shot 2020-05-20 at 10 22 33 AM

#### New:

Screen Shot 2020-05-20 at 10 27 44 AM

**Comment by gdoyle1** [comment]
@mcoker @evwilkin I wasn't aware of the dropdown not using the small sized caret. I switched that example up with our new labels instead so we don't have to worry about that. I ended up zipping together screenshots of what we can use per icon size and attached them here!
[IconPageScreenShots.zip](https://github.com/patternfly/patternfly-org/files/4657882/IconPageScreenShots.zip)

I have a couple comments on the responsiveness of the page:
1. Can we drop the icon sizes image below the listing of the icon sizes? (Right now it is dropping below everything)
2. I'm seeing some weird responsiveness stuff here with the alignment image - we can just drop this image below the text
Screen Shot 2020-05-20 at 11 47 22 AM

Everything @mcoker said above covers anything else I was seeing with the table!

@mcoker To answer your question about the fa regular icons - we provide a link back to this page: https://www.patternfly.org/v4/get-started/developers#using-styles for that. We need to add some more text there for:
1. Fixing the part that says "overpass"
2. Adding the part about "A user would have to bundle their own font-awesome regular package for this to work since we only ship the font-awesome solid webfont"

Is that something we should open another issue for?

**Comment by bdellasc** [comment]
The page is looking great! I have some feedback:

> A user would have to bundle their own font-awesome regular package for this to work since we only ship the font-awesome solid webfont. Is that noted anywhere?

I was under the impression that the only limitation we had for Font Awesome was whether or not it was in the "free" set. I didn't think it mattered if it was a solid or regular icon. Was there a reason why we're not including the regular icons? That will really limit our options. Two icons marked as regular (far) on the test page are showing the solid versions in the table (the other regular icons seem ok. Here are the two showing the solid versions: 
- fa-clock
- fa-comments

The regular versions of these icons used in the design kit, so there's a mismatch. There aren't many we use, but they are part of the recommendations.

**New Recommendations table:**

- Can we keep the icon names from wrapping in this table as well? They're very clean when not wrapping in the top table.

- row for "fas fa-plus-square" - the second line in the contextual usage text is actually referring to the outlined version of the icon (second icon). 

- Can we vertically align that second line of text with the second icon? If not, we need to modify the second line of text to say: "Use 'pf-icon-add-circle-o' if there are many instances of the icon in a UI (data list, table, etc) to reduce visual noise."

- For the 'pf-icon-unplugged' row, can we use 'n/a' instead of a '-' in the new recommendation column?

- For pf-icon-help row, can we split the contextual text so the outline version is used for contextual help and the solid is for a help system in the masthead?:
(vertically aligned with the first line of text) "Indicates the availability of contextual help"  
(vertically aligned with the second line of text) "Indicates the availability of a help system, typically in the masthead"

 If not, we need to modify the text to say: " 'far fa-question-circle' indicates the availability of contextual help, while 'fas fa-question-circle' indicates the availability of a help system in the masthead.

Thanks for all the hard work on this page! It's looking great!

File: packages/v4/src/content/design-guidelines/styles/icons/IconRecommendations.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        const { icon, name, reactIcon } = cellLine;
+        const Icon = reactIcon !== 'svg' && icons[reactIcon];
+        const iconSvg = reactIcon === 'svg' && <img src={iconSvgs[name]} className="ws-icon-svg" alt={`${icon} icon`} />;
+        
+        (columnName === 'iconUsage')
+          ? cellObj.title = cellLine
+          : cellObj.title.push(<div className={`${css(styles.modifiers.fitContent)} ws-recommendations-entry`} key={`${name}-${index}`}><span className="ws-recommendations-icon">
+            {Icon && <Icon />}
+            {iconSvg && iconSvg}
+            {}
```
[the remark this answers is not in the document]
Comment (evwilkin) [reply packages/v4/src/content/design-guidelines/styles/icons/IconRecommendations.js:59]: Removed - thanks!

File: packages/v4/src/content/design-guidelines/styles/icons/IconRecommendations.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      rowObj[columnName].map((cellLine, index) => {
+        const { icon, name, reactIcon } = cellLine;
+        const Icon = reactIcon !== 'svg' && icons[reactIcon];
+        const iconSvg = reactIcon === 'svg' && <img src={iconSvgs[name]} className="ws-icon-svg" alt={`${icon} icon`} />;
+        
+        (columnName === 'iconUsage')
+          ? cellObj.title = cellLine
+          : cellObj.title.push(<div className={`${css(styles.modifiers.fitContent)} ws-recommendations-entry`} key={`${name}-${index}`}><span className="ws-recommendations-icon">
+            {Icon && <Icon />}
+            {iconSvg && iconSvg}
```
[the remark this answers is not in the document]
Comment (evwilkin) [reply packages/v4/src/content/design-guidelines/styles/icons/IconRecommendations.js:58]: Updated

File: packages/v4/src/content/design-guidelines/styles/icons/IconsTable.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+  onSort = (_event, index, direction) => {
+    this.setState({
+      sortBy: {
+        index,
+        direction
+      }
+    });
+  }
+
+  onDownloadSvg = ({ currentTarget }) => {
```
[the remark this answers is not in the document]
Comment (evwilkin) [reply packages/v4/src/content/design-guidelines/styles/icons/IconsTable.js:64]: The copy above the table speaks towards this:

> Click on any single pf-icon in the table to download it as an SVG.

File: packages/v4/src/content/design-guidelines/styles/icons/recommendations.js [hunk 4]
```diff
@@ -0,0 +1,1339 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (evwilkin) [reply packages/v4/src/content/design-guidelines/styles/icons/recommendations.js:1]: React imported due to JSX used on line 1331.  Renamed to `icon-migrations.js` - thanks for the naming suggestion

File: packages/v4/src/content/design-guidelines/styles/icons/iconsData.js [hunk 5]
```diff
@@ -0,0 +1,1199 @@
+export const iconsData = [
```
[the remark this answers is not in the document]
Comment (evwilkin) [reply packages/v4/src/content/design-guidelines/styles/icons/iconsData.js:1]: Renamed file

**Comment by rachael-phillips** [comment]
Screen Shot 2020-05-22 at 3 14 30 PM

Is this "design-guidelines" at the top going to go away?

**Comment by evwilkin** [comment]
> Is this "design-guidelines" at the top going to go away?

@rachael-phillips removed this, thanks!

@gdoyle1 I've updated the responsive breakpoint.  The broken links should work as expected when live, this is a known issue in Org with the preview site combined with internal relative links.

**Comment by bdellasc** [comment]
Just verifying that all the fixes were completed based on my comment from last week. Great work everyone, it looks fantastic! Thanks!

**Pull request merged**